tests/e2e/vmlistener_test.go

```diff
@@ -104,34 +116,85 @@ func (s *VMListenerTestParams) TestServiceRestart(c *C) {
 func (s *VMListenerTestParams) TestKillVM(c *C) {
 	misc.LogTestStart(c.TestName())
 
-	// make sure vm was powered on
-	powerState := govc.GetVMPowerState(s.dockerHostName)
-	log.Printf("VM[%s]'s current power state is [%s]", s.dockerHostName, powerState)
-	c.Assert(powerState, Equals, properties.PowerOnState, Commentf("VM [%s] should be powered on state", s.dockerHostName))
+	// Kill the VM
+	isVMKilled := killVM(s.esx, s.vm1Name)
+	c.Assert(isVMKilled, Equals, true, Commentf("VM [%s] was not killed successfully", s.vm1Name))
+
+	// Status should be detached
+	volStatus := verification.GetVMAttachedToVolUsingAdminCli(s.volumeName, s.esx)
+	c.Assert(volStatus, Equals, properties.DetachedStatus, Commentf("Volume %s is still attached", s.volumeName))
+
+	// Power on VM
+	govc.PowerOnVM(s.vm1Name)
+	isStatusChanged := misc.WaitForExpectedState(govc.GetVMPowerState, s.vm1Name, properties.PowerOnState)
+	c.Assert(isStatusChanged, Equals, true, Commentf("VM [%s] should be powered on state", s.vm1Name))
+
+	// Status should be detached
+	status := verification.VerifyDetachedStatus(s.volumeName, s.vm1, s.esx)
+	c.Assert(status, Equals, true, Commentf("Volume %s is still attached", s.volumeName))
 
-	// grab worldID/vmProcessID
-	processID := esxcli.GetVMProcessID(s.esxIP, s.dockerHostName)
-	log.Printf("VM's process ID is: %s", processID)
+	misc.LogTestEnd(c.TestName())
+}
 
-	// kill VM
-	isVMKilled := esxcli.KillVM(s.esxIP, processID)
-	c.Assert(isVMKilled, Equals, true, Commentf("Unable to kill VM %s ...", s.dockerHostName))
+// Test failover by killing a VM and then mount the volume to a different VM
+// 1. Setup: Create a volume
+// 2. Setup: Attach it to a container
+// 3. Kill the VM
+// 4. Verification: volume status should be detached
+// 5. Attach it to a different container
```

Comment:
I think it should be attach to a container on different VM.
Thats what it looks like from line 157